src/org/rascalmpl/library/String.rsc

```diff
@@ -534,11 +536,37 @@ public java str capitalize(str src);
 @javaClass{org.rascalmpl.library.Prelude}
 public java str uncapitalize(str src);
 
+@synopsis{Base-64 encode the characters of a string.}
+@description{
+   Convert the characters of a string to a list of bytes using UTF-8 encoding and then encode these bytes using base-64 encoding
+   as defined by RFC 4648: https://www.ietf.org/rfc/rfc4648.txt.
+}
+@javaClass{org.rascalmpl.library.Prelude}
+public java str toBase64(str src, str charset=DEFAULT_CHARSET);
+
+@synopsis{Decode a base-32 encoded string.}
+@description {
+  Convert a base-32 encoded string to bytes and then convert these bytes to a string using the specified cahracter set.
+  The base-32 encoding used is defined by RFC 4648: https://www.ietf.org/rfc/rfc4648.txt.
+}
 @javaClass{org.rascalmpl.library.Prelude}
-public java str toBase64(str src);
+public java str fromBase64(str src, str charset=DEFAULT_CHARSET);
 
+@synopsis{Base-32 encode the characters of a string.}
+@description{
+   Convert the characters of a string to a list of bytes using UTF-8 encoding and then encode these bytes using base-32 encoding
+   as defined by RFC 4648: https://www.ietf.org/rfc/rfc4648.txt.
+}
+@javaClass{org.rascalmpl.library.Prelude}
+public java str toBase32(str src, str charset=DEFAULT_CHARSET, bool includePadding=true);
```

Comment:
it's good that we have the optional padding, but don't we want that for the other cases as well?

Reply:
For the `toBase64` method I totally agree. For files this seems less important somehow.